textalyze.js

```diff
@@ -1,23 +1,30 @@
-/**
- * This is the base code for v0.1 of our JavaScript text analyzer.
- * Visit https://github.com/jfarmer/text-analysis to see what to do.
- *
- * Send a message in Slack if you're stuck or unsure what to do.  These
- * comments are here to help you, but please delete them as you go along.
- */
-
 /**
  * Given an input Array, returns a Map containing the count of each item in the input.
  * @param {Array} array - The array of items to count
  * @param {Map} counts - A Map containing the counts of the items in the input array
  */
 function itemCounts(array) {
-  let counts = new Map();
+  const counts = new Map();
```

Comment:
Exercise: how would you write this as a single call to [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)?

Reply:
Oops, left this comment in the wrong spot.

Reply:
I would do like it

```
const counts = array.reduce((map, item) => {
    const count = map.get(item) || 0;
    return map.set(item, count + 1);
  }, new Map());
```